include/rcutils/shared_library.h

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCUTILS__SHARED_LIBRARY_H_
+#define RCUTILS__SHARED_LIBRARY_H_
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <string.h>
+
+#ifndef _WIN32
+#include <dlfcn.h>
+#else
+#include <windows.h>
+#endif  // _WIN32
+
+#include "rcutils/macros.h"
+#include "rcutils/visibility_control.h"
+
+/// The structure
+typedef struct rcutils_shared_library_t
+{
+  /// The pointer to the shared library.
+  #ifndef _WIN32
+  void * lib_pointer;
+  #else
+  HINSTANCE lib_pointer;
+  #endif
+  /// The path of the shared_library
+  char * library_path;
+} rcutils_shared_library_t;
+
+/// Return shared library pointer.
+/**
+ * \param[in] library_path path to the library
```

Comment:
It would be nice to know:

- does this string get copied into the returned structure?
  - if so how is it allocated?
- how is the returned structure allocated?